app/src/androidTest/java/ch/epfl/cs311/wanderwave/viewmodel/ProfileViewModelTest.kt

```diff
@@ -90,20 +92,38 @@ class ProfileViewModelTest {
   fun retrieveTrackTest() = runBlockingTest {
     val track = ListItem("bbbb", "bbbb", null, "bbbb", "bbbb", false, true)
     val track2 = ListItem("aaaa", "aaaaa", null, "aaaaa", "aaaaa", false, true)
+
+    // Mocking responses for your spotifyController
     every { spotifyController.getAllElementFromSpotify() } returns flowOf(listOf(track))
-    every { spotifyController.getAllChildren(track) } returns
-        flowOf(listOf(track2)) // No empty list first
+    every { spotifyController.getAllChildren(track) } returns flowOf(listOf(track2))
 
     viewModel.createSpecificSongList("TOP_SONGS")
-    viewModel.retrieveTracks()
+
+    // Start observing the Flow before triggering actions that modify it
+    val job = launch {
+      viewModel.songLists.collect { songLists ->
+        if (songLists.isNotEmpty()) {
+          cancel() // Stop collecting once we have our expected result
+        }
+      }
+    }
+
+    // Trigger the operations that will cause the song lists to be populated
+    viewModel.retrieveTracks(this)
+    // Wait for the job to complete which includes Flow collection
+    job.join()
+
+    // Since Flow collection is asynchronous, ensure the Flow has time to collect
     advanceUntilIdle()
-    var songLists = viewModel.songLists.value
-    assertFalse("Song list should not be empty after adding a track", songLists.isEmpty())
 
-    songLists = viewModel.songLists.value
-    // Check if the track was added correctly
-    val songsInList = songLists.find { it.name == "TOP SONGS" }?.tracks ?: emptyList()
-    Log.d("Temp", songsInList.toString())
+    // Optionally check additional conditions after ensuring Flow had time to collect
+    assertFalse(
+        "Song list should not be empty after adding a track", viewModel.songLists.value.isEmpty())
+    Log.d("fwefewfew", viewModel.songLists.value.first().tracks.toString())
```

Comment:
Use copilot for this, it's very usefull to have the logs having the right title and not something random like this. Can you change it ?